code/datums/mood_events/generic_positive_events.dm

```diff
@@ -19,6 +19,14 @@
 /datum/mood_event/besthug/add_effects(mob/friend)
 	description = "<span class='nicegreen'>[friend.name] is great to be around, [friend.p_they()] makes me feel so happy!</span>\n"
 
+/datum/mood_event/warmhug
+	description = "<span class='nicegreen'>Warm cozy hugs are the best!</span>\n"
+	mood_change = 1
+	timeout = 2 MINUTES
+
+/datum/mood_event/warmhug/add_effects()
+	description = "<span class='nicegreen'>Someone gave me a warm cozy hug!</span>\n"
```

Comment:
what's this for?

Reply:
When you click the moodlet icon it reports this back to you. Am I wrong?

Reply:
that's what the description in **/datum/mood_event/warmhug** does

You only need this proc if you, say, want the message to include the name of the person that hugged you, like in **/datum/mood_event/besthug**